internal/services/compute/image_resource.go

```diff
@@ -443,6 +471,12 @@ func flattenImageOSDisk(input *images.ImageStorageProfile) []interface{} {
 				"size_gb":                diskSizeGB,
 				"disk_encryption_set_id": diskEncryptionSetId,
 			})
+
+			if v.StorageAccountType != nil && features.FourPointOhBeta() {
+				output = append(output, map[string]interface{}{
+					"storage_type": string(*v.StorageAccountType),
+				})
+			}
```

Comment:
I don't think this will work as expected, since the intended output of this helper func should be a single `map[string]interface{}` containing all properties (contained as the only element of a `[]interface{}`

Due to GH limitations I can't suggest inline, but something like this should work:

```go
			properties := map[string]interface{}{
				"blob_uri":               blobUri,
				"caching":                caching,
				"managed_disk_id":        managedDiskId,
				"os_type":                string(v.OsType),
				"os_state":               string(v.OsState),
				"size_gb":                diskSizeGB,
				"disk_encryption_set_id": diskEncryptionSetId,
			}

			if v.StorageAccountType != nil && features.FourPointOhBeta() {
				properties["storage_type"] = string(*v.StorageAccountType)
			}

			output = append(output, properties)
```

Reply:
done